backend/src/test/java/ai/verta/modeldb/blobs/DiffAndMerge.java

```diff
@@ -28,6 +28,27 @@ public void diffAndMerge(Blob a, Blob b) throws ModelDBException {
     assertEquals(newB, diffedB);
   }
 
+  @Property
+  public void diffAndMergeNullCheck(Blob a) throws ModelDBException {
+    Blob newA = enforceOneof(a);
+    BlobDiff d = DiffComputer.computeBlobDiff(newA, null);
+
+    // Applying the diff on top of the original A should get original B
+    Blob diffedB = DiffMerger.mergeBlob(newA, d);
+    assertNull(diffedB);
+
+    // Reapplying the diff should not change the result
+    diffedB = DiffMerger.mergeBlob(diffedB, d);
+    assertNull(diffedB);
+  }
+
+  @Property
+  public void diffAndMergeForSameBlob(Blob a) throws ModelDBException {
+    Blob newA = enforceOneof(a);
+    BlobDiff d = DiffComputer.computeBlobDiff(newA, newA);
+    assertNull(d);
```

Comment:
here expected is a null result for BlobDiff result but in some cases, it will return not null results. 
for the `newA` JSON value is 
``` 
dataset {
  s3 {
    components {
      path {
        path: "cZEhT"
        size: 17452581301193614155
        sha256: "]Mo@P"
      }
    }
    components {
      path {
        size: 9563800031935968269
        last_modified_at_source: 7682457601031712354
        sha256: "S]mw3"
        md5: "0/B1w"
      }
    }
    components {
      path {
        path: "RsFiq"
        size: 16727320813202495771
        last_modified_at_source: 12906393968766738848
        md5: "4cZYF"
      }
    }
    components {
      path {
        path: "MV]W|"
        last_modified_at_source: 1064482319913550563
        sha256: "xe23}"
        md5: "haRgQ"
      }
    }
    components {
      path {
        size: 7509297217166798670
        last_modified_at_source: 9280711701201284371
        sha256: "S@oIQ"
        md5: "J;o]w"
      }
    }
  }
}
```
And Return BlobDiff value is 
``` 
dataset {
  s3 {
    components {
      path {
        status: DELETED
        A {
          size: 7509297217166798670
          last_modified_at_source: 9280711701201284371
          sha256: "S@oIQ"
          md5: "J;o]w"
        }
      }
    }
    components {
      path {
        status: DELETED
        A {
          size: 9563800031935968269
          last_modified_at_source: 7682457601031712354
          sha256: "S]mw3"
          md5: "0/B1w"
        }
      }
    }
    components {
      path {
        status: ADDED
        B {
          size: 9563800031935968269
          last_modified_at_source: 7682457601031712354
          sha256: "S]mw3"
          md5: "0/B1w"
        }
      }
    }
    components {
      path {
        status: ADDED
        B {
          size: 7509297217166798670
          last_modified_at_source: 9280711701201284371
          sha256: "S@oIQ"
          md5: "J;o]w"
        }
      }
    }
  }
}
```
which is wrong because we have passed the same `newA` in diff as parameter.

Reply:
Discussion: https://github.com/VertaAI/modeldb/pull/360#discussion_r394496935